### ISSUE TITLE
ShouldBeEquivalentTo flags two seemingly equivalent objects

### ISSUE THREAD

**Issue by godrose** [fluentassertions/fluentassertions#973]
### Description

Version 5.5.1 fails existing test - used to to work in 5.4.2

### Complete minimal example reproducing the issue

Checkout the following repo: https://github.com/LogoFX/logofx-client-mvvm-viewmodel
Run the Unit tests - all tests are green
Upgrade to 5.5.1
Run the Unit tests - now some are red.

### Expected behavior:

Tests remain green

### Actual behavior:

Some tests are red

### Versions

- [x] FA 5.5.1
- [x] .NET Framework 4.6.1

**Comment by godrose** [comment]
@jnyrup Thanks for the reply.
Please see whether this example is more suitable for you
https://github.com/godrose/FailingTest

The only test fails with the following message:
`
FluentAssertions.Execution.AssertionFailedException
Expected args.OldItems to be {System.Object (HashCode=45523402)}, but found {System.Object (HashCode=45523402)}.
`

Using previous version (5.4.2) restores the green state of the test.

**Comment by jnyrup** [comment]
I don't have time to look deeper into this today, but I reduced the test case to this:

```c#
object item = new object();
object[] array = new[] { item };

IList readOnlyList = ArrayList.ReadOnly(array);
readOnlyList.Should().BeEquivalentTo(array);
```

**Comment by jnyrup** [comment]
@dennisdoomen If you haven't already located the it, `git bisect` marks b3b0245a56d1808b68d6fdbbf0ce95073e93d06a as the commit that changed behavior.

**Comment by dennisdoomen** [comment]
Fixed in 5.5.2
